src/overlays/actors/ovl_Bg_Spot06_Objects/z_bg_spot06_objects.c

```diff
@@ -29,39 +64,440 @@ const ActorInit Bg_Spot06_Objects_InitVars = {
+                this->collider.list[0].dim.worldSphere.center.z = thisx->posRot.pos.z + 16.0f;
+            } else {
+                this->actionFunc = BgSpot06Objects_LockWait;
+                this->collider.list[0].dim.worldSphere.radius = this->collider.list[0].dim.modelSphere.radius;
+                this->collider.list[0].dim.worldSphere.center.z = thisx->posRot.pos.z;
+            }
+
+            this->collider.list[0].dim.worldSphere.center.x = thisx->posRot.pos.x;
+            this->collider.list[0].dim.worldSphere.center.y = thisx->posRot.pos.y;
+            thisx->colChkInfo.mass = -1;
+            break;
+        case LHO_WATER_PLANE:
+            Actor_ProcessInitChain(thisx, sInitChainWaterPlane);
+            thisx->flags = 0x30;
+
+            if (LINK_IS_ADULT && !(gSaveContext.eventChkInf[6] & 0x200)) {
+                if (gSaveContext.sceneSetupIndex < 4) {
+                    this->lakeHyliaWaterLevel = -681.0f;
+                    globalCtx->colCtx.colHeader->waterBoxes[LHWB_GERUDO_VALLEY_RIVER_LOWER].ySurface =
+                        WATER_LEVEL_RIVER_LOWERED;
+                    globalCtx->colCtx.colHeader->waterBoxes[LHWB_GERUDO_VALLEY_RIVER_LOWER].zMin -= 50;
+                    globalCtx->colCtx.colHeader->waterBoxes[LHWB_MAIN_1].ySurface = WATER_LEVEL_LOWERED;
+                    globalCtx->colCtx.colHeader->waterBoxes[LHWB_MAIN_2].ySurface = WATER_LEVEL_LOWERED;
+                    this->actionFunc = BgSpot06Objects_DoNothing;
+                } else {
+                    thisx->posRot.pos.y = this->lakeHyliaWaterLevel = -681.0f;
+                    thisx->posRot.pos.y += (f32)WATER_LEVEL_RAISED;
+                    this->actionFunc = BgSpot06Objects_WaterPlaneCutsceneWait;
+                }
+            } else {
+                this->lakeHyliaWaterLevel = 0.0f;
+                this->actionFunc = BgSpot06Objects_DoNothing;
+            }
+            break;
+        case LHO_ICE_BLOCK:
+            Actor_ProcessInitChain(thisx, sInitChain);
+            DynaPolyActor_Init(thisx, DPM_UNK);
+            CollisionHeader_GetVirtual(&D_06001238, &colHeader);
+            this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, thisx, colHeader);
+            this->actionFunc = BgSpot06Objects_DoNothing;
+
+            if (LINK_IS_CHILD) {
+                Actor_Kill(thisx);
+            }
+            break;
+    }
+}
+
+void BgSpot06Objects_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot06Objects* this = THIS;
+
+    switch (this->dyna.actor.params) {
+        case LHO_WATER_TEMPLE_ENTRACE_GATE:
+        case LHO_ICE_BLOCK:
+            DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+            break;
+        case LHO_WATER_TEMPLE_ENTRANCE_LOCK:
+            Collider_DestroyJntSph(globalCtx, &this->collider.base);
+            break;
+        case LHO_WATER_PLANE:
+            break;
+    }
+}
+
+/*
```

Comment:
Multi-line comments should ideally start with `/**` (relevant in the rest of the file)
```suggestion
/**
```